sources/TileDB.CSharp/FragmentInfo.cs

```diff
@@ -651,7 +651,7 @@ private static void ValidateDomainType<T>(DataType dataType)
             {
                 ThrowHelpers.ThrowStringTypeMismatch(dataType);
             }
-            if (dataType != EnumUtil.TypeToDataType(typeof(T)))
+            if (EnumUtil.DataTypeToType(dataType) != typeof(T))
```

Comment:
This fixes a bug in our type checking logic that manifested when using date/time types in the dimension labels example.

There is a many-to-one mapping between a TileDB data type and a .NET type. `long` maps by default to `TILEDB_INT64` but there are other possible TileDB data types that map to `long`, such as `TILEDB_DATETIME_*S`. We were previously "widening" the .NET type into its default TileDB data type, which caused mismatches. The correct way is to "narrow" the TileDB data type into the .NET type.

Reply:
Fixed in #291.